scripts/test-gem-build

```diff
@@ -0,0 +1,32 @@
+#! /usr/bin/env bash
+#
+#  run as part of CI
+#
+if [[ $# -lt 2 ]] ; then
+  echo "usage: $(basename $0) <output_dir> <platform>"
+  exit 1
+fi
+
+set -e -u
+
+OUTPUT_DIR=$1
+BUILD_NATIVE_GEM=$2
+
+test -e /etc/os-release && cat /etc/os-release
+
+set -x
+
+bundle install --local || bundle install
+bundle exec rake set-version-to-timestamp
```

Comment:
I think I asked this before but why do we need to use a timestamped version rather than, say, using `2.0.0.beta1` or whatever the current version is in `re2.gemspec`? Is it because the gems will be uploaded as artifacts and it might be misleading to people browsing the Actions runs?

Reply:
> Is it because the gems will be uploaded as artifacts and it might be misleading to people browsing the Actions runs?

Yes, and if you're testing locally you avoid the issue of overwriting a properly-installed gem.

Reply:
I wonder if we could simplify things by explicitly passing the version during the build. By default, it could be the `ref_name` plus the `run_number` but we could use [an input for `workflow_dispatch`](https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/) that overrides that so I could, e.g., manually trigger a build for 2.0.0.beta1.

Alternatively, if we didn’t add and rely on `RE2::VERSION`, maybe we could get away with making the gemspec set the version from `ENV.fetch("RE2_VERSION", "2.0.0.beta1)` (or whatever the expected version should be) so we can override it for CI runs but also use it for releases.

Reply:
Yeah, that would be a nice improvement.